src/storage/src/storage_state/async_storage_worker.rs

```diff
@@ -92,11 +184,73 @@ impl<T: Timestamp + Lattice + Codec64> AsyncStorageWorker<T> {
+                                    &persist_clients,
+                                    &ingestion_description,
+                                    &resume_upper,
+                                )
+                                .await;
+                                upper.into_iter().map(|ts| ts.encode_row()).collect()
+                            }
+                            GenericSourceConnection::Postgres(_) => {
+                                let upper = reclock_resume_frontier::<PostgresSourceConnection, _>(
+                                    &persist_clients,
+                                    &ingestion_description,
+                                    &resume_upper,
+                                )
+                                .await;
+                                upper.into_iter().map(|ts| ts.encode_row()).collect()
+                            }
+                            GenericSourceConnection::S3(_) => {
+                                let upper = reclock_resume_frontier::<S3SourceConnection, _>(
+                                    &persist_clients,
+                                    &ingestion_description,
+                                    &resume_upper,
+                                )
+                                .await;
+                                upper.into_iter().map(|ts| ts.encode_row()).collect()
+                            }
+                            GenericSourceConnection::LoadGenerator(_) => {
+                                let upper =
+                                    reclock_resume_frontier::<LoadGeneratorSourceConnection, _>(
+                                        &persist_clients,
+                                        &ingestion_description,
+                                        &resume_upper,
+                                    )
+                                    .await;
+                                upper.into_iter().map(|ts| ts.encode_row()).collect()
+                            }
+                            GenericSourceConnection::TestScript(_) => {
+                                let upper =
+                                    reclock_resume_frontier::<TestScriptSourceConnection, _>(
+                                        &persist_clients,
+                                        &ingestion_description,
+                                        &resume_upper,
+                                    )
+                                    .await;
+                                upper.into_iter().map(|ts| ts.encode_row()).collect()
+                            }
+                        };
+
                         let res = response_tx.send(
                             AsyncStorageWorkerResponse::IngestDescriptionWithResumeUpper(
                                 id,
                                 ingestion_description,
                                 resume_upper,
+                                source_resume_upper,
```

Comment:
can you add a comment here explaining why this is `Row`-encoded, and that in the future we want ingestion to be fully generic?